### PR TITLE
CI: Add e2e to CI pipeline

### PR DESCRIPTION
**Pull request by eduser25**
Will run all tests under e2e folder. Failfast is enabled as these can potentially be slow.
Also added proper CTR_TAG reading on test framework.

patchset[]:
- Increased all static timeouts for most pod bringup and some test conditions (as seen pod bringup could sometimes take a lot of time).
Future efforts will include parametrization (#1825)

**Affected area**:

- Tests                  [X]
- CI System          [X]

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

**Comment by eduser25** [comment]
> Integration Test with jetstack/cert-manager, SMI traffic policies (no Traffic Split), egress disa...
failed 4 minutes ago in 14m 50s

Seems unrelated but.... I wonder changes are causing some side effect. Will trigger another run to see.

**Comment by eduser25** [comment]
the failing batch from yesterday night had something to do with Github, both demo and E2E had failures with unrelated OSM pods not coming up.

**Review by shashankram — APPROVED** [review]
Is there a plan to remove duplicate tests run by existing CI?

File: tests/e2e/common.go [hunk 1]
```diff
@@ -79,13 +79,19 @@ func registerFlags(td *OsmTestData) {
...
 	flag.StringVar(&td.ctrRegistryPassword, "ctrRegistrySecret", os.Getenv("CTR_REGISTRY_PASSWORD"), "Container registry secret")
 
-	flag.StringVar(&td.osmImageTag, "osmImageTag", "latest", "OSM image tag")
+	flag.StringVar(&td.osmImageTag, "osmImageTag", func() string {
+		tmp := os.Getenv("CTR_TAG")
+		if len(tmp) != 0 {
+			return tmp
+		}
+		return "latest"
+	}(), "OSM image tag")
```
Comment (draychev) [inline tests/e2e/common.go:94]: We could move this https://github.com/openservicemesh/osm/blob/c98e2556ca03ec7a0774e38066ece4a16f53ebb2/demo/cmd/common/books.go#L96-L103 into some util package and use it.

**Review by draychev — APPROVED** [review]
Thank you!

File: tests/e2e/common.go (hunk 1 above)
Comment (eduser25) [reply tests/e2e/common.go:94]: makes sense
will do

**Comment by eduser25** [comment]
> Is there a plan to remove duplicate tests run by existing CI?

Wait, which ones? Referring to demo/maestro?

**Comment by shashankram** [comment]
> > Is there a plan to remove duplicate tests run by existing CI?
> 
> Wait, which ones? Referring to demo/maestro?

Yes, if we are testing the different scenarios run by maestro (traffic target, split, different certificate providers, egress etc.) in the e2e tests, it might make sense to not test the same stuff twice in the interest of CI time.

**Comment by eduser25** [comment]
> > > Is there a plan to remove duplicate tests run by existing CI?
> > 
> > 
> > Wait, which ones? Referring to demo/maestro?
> 
> Yes, if we are testing the different scenarios run by maestro (traffic target, split, different certificate providers, egress etc.) in the e2e tests, it might make sense to not test the same stuff twice in the interest of CI time.

Absolutely, I did not visit this yet. If you folks feel that E2E coverage is good enough, we can disable maestro anytime.
I think @nojnhuh is wrapping up egress tests. I'll check what are we missing.

**Pull request merged**